### PR TITLE
Standardize Form Input Across the App

### PR DESCRIPTION
**Pull request by jameskerr**
Five new common components have been added:

* TextInput
* SelectInput
* FileInput
* InputLabel
* InputField

They all look something like this: 

image

These have replaced all the form components across the app.

This is in preparation for implementing issue #849

### CODE REVIEW

File: src/js/components/DebugModal.js [hunk 1]
```diff
@@ -35,15 +36,15 @@ function DebugModalContents() {
...
+      <InputField>
+        <InputLabel>Query:</InputLabel>
+        <TextInput
+          autoFocus
           value={program}
           onChange={(e) => setProgram(e.target.value)}
           {...reactElementProps("debugProgram")}
         />
-      </Form>
+      </InputField>
```
Comment (jameskerr) [inline src/js/components/DebugModal.js:47]: Screen Shot 2020-06-22 at 12 50 08 PM

File: src/js/components/SpaceModal.js [hunk 2]
```diff
@@ -42,14 +44,17 @@ export default function SpaceModal() {
...
+        <InputLabel>New Name:</InputLabel>
+        <TextInput
+          autoFocus
+          label="Space Name"
+          value={value}
+          onChange={(e) => {
+            onChange(e.target.value)
+          }}
+        />
+      </InputField>
```
Comment (jameskerr) [inline src/js/components/SpaceModal.js:57]: Screen Shot 2020-06-22 at 12 49 37 PM

File: src/css/_time-span-pickers.scss [hunk 3]
```diff
@@ -24,6 +24,7 @@
 
   input {
     @include label-small;
+    @include input-focus;
```
Comment (jameskerr) [inline src/css/_time-span-pickers.scss:27]: Makes editing the time span look like this: 
Screen Shot 2020-06-22 at 12 33 16 PM

File: src/js/components/SpaceModal.js (hunk 2 above)
Comment (mason-fish) [reply src/js/components/SpaceModal.js:57]: I think that `label="Space Name"` prop/attribute in the `<TextInput />` will not be of use anymore.

Comment (mason-fish) [reply src/js/components/SpaceModal.js:57]: Also, I'm aware that putting colons after labels is helpful for older screen readers to parse forms but from a design perspective I find I prefer these labels only with the emphasis we are already giving them (boldness, placement, followed by a clearly editable box) and no colon. Just one person's opinion though :)

File: src/css/settings/_colors.scss [hunk 4]
```diff
@@ -43,8 +43,20 @@ $pink: #ad3f95;
 
 $purple: #544aa6;
 $orange: #eeb457;
-$aqua: #252525;
-$slate: #737373;
+
+$azure: #2f629c;
```
[the remark this answers is not in the document]
Comment (jameskerr) [reply src/css/settings/_colors.scss:47]: Thanks for bringing this up. My first inclination is to hold onto my scss files. I enjoy writing scss, it's super stable (12 years old) and simple, and the feedback loop is super short (due to livereload). 

However, I was skeptical of when html and javascript were blended together with react, but that turned out to be a great idea. Blending the css into the component is probably going to be a good idea too. The fact that you've already used it and really like it is a plus too.

Ok, I'm willing do an experiment for a while where every new component I make will use StyledComponents.


Some articles I read that influenced me: 

https://medium.com/building-crowdriff/styled-components-to-use-or-not-to-use-a6bb4a7ffc21

https://blog.logrocket.com/moving-from-scss-to-styled-components-advantages-and-caveats/

**Pull request merged**